pkgs/by-name/st/stylance-cli/package.nix

```
@@ -2,17 +2,26 @@
   lib,
   rustPlatform,
   fetchCrate,
+  testers,
 }:
-rustPlatform.buildRustPackage rec {
+let
   pname = "stylance-cli";
-  version = "0.7.1";
+  version = "0.7.4";
+in
+rustPlatform.buildRustPackage (finalAttrs: {
+  inherit pname version;
 
   src = fetchCrate {
     inherit pname version;
-    hash = "sha256-Cdv+Lz+l0+8Jdk9stHACXDbUPedM/YryDMExdsqVvsU=";
+    hash = "sha256-lGgKmNqZ0nflVAM3GMDwGgxnXyLCqVz1bTUsvabXmj8=";
   };
 
-  cargoHash = "sha256-cwgR5AHCeS9YkaJlyFxvEOrBXg7/tXNGXgtSEPHAwm4=";
+  cargoHash = "sha256-HWZQNEKTyNnmA1twN5cfo5RY2tOeCnL6zEp+M4F+Tqg=";
+
+  passthru.tests.version = testers.testVersion {
```

We usually prefer `versionCheckHook` now as it eliminates one more derivation, and always fails loudly. `testers.testVersion` can still be used if the intention is to get rid of the influence of build environment, which usually isn't the case for rust.

---

Thanks, I didn't know about `versionCheckHook`.